Towny/src/main/java/com/palmergames/bukkit/towny/utils/SpawnUtil.java

```diff
@@ -669,12 +674,52 @@ private static void initiateSpawn(Player player, Location spawnLoc, int cooldown
 			// Don't use teleport warmup
 			if (player.getVehicle() != null)
 				player.getVehicle().eject();
-			PaperLib.teleportAsync(player, spawnLoc, TeleportCause.COMMAND);
+			
+			List<Entity> pets = getPets(player);
+			pets.add(player);
+			teleportEntities(spawnLoc, pets, TeleportCause.COMMAND);
 			if (cooldown > 0 && !hasPerm(player, PermissionNodes.TOWNY_SPAWN_ADMIN_NOCOOLDOWN))
 				CooldownTimerTask.addCooldownTimer(player.getName(), "teleport", cooldown);
 		}
 	}
 
+	/**
+	 * Gets any pets owned by the specified player that are nearby them
+	 * 
+	 * @param player The player to get the pets of
+	 */
+	public static List<Entity> getPets(Player player) {
+		List<Entity> pets = new ArrayList<>();
+		
+		for (Entity entity : player.getNearbyEntities(16, 16, 16)) {
+			if (!(entity instanceof Tameable tameable)) continue;
+
+			AnimalTamer tamer = tameable.getOwner();
+			if (tamer == null) continue;
+
+			if (!(entity instanceof Sittable sittable)) continue;

```

Comment:
What about non sittable pets? When #6865 is merged the types of entities that are teleported could be made configurable

Reply:
what pets are non-sittable that also follow you like a dog or cat

Reply:
or do you mean for adding things like horses, camels etc.